### PR TITLE
Added linked locale messages feature.

### PR DESCRIPTION
**Pull request by HatEmU**
I also added tests for the stuff I made and everything works fine :)

To further improve this feature we'll need to add recursion detection.

### CODE REVIEW

**Comment by HatEmU** [comment]
This is PR that addresses the issue #279

**Comment by HatEmU** [comment]
After a bit of debugging I found out that nested named arguments don't workif you apply modifiers to them.

For example:

`
{
    'nestedNamedArg': 'this is a @.lower:namedArg',
    'namedArg': 'nested {namedArg}'
}
`

And if you then do this:

`Localization.instance.tr('nestedNamedArg', namedArgs: {'namedArg': 'Argument'});`

It will fail to remove the brackets and replace the key and will instead apply formatting to the key like so:

"this is a {namedarg}"

**Comment by HatEmU** [comment]
Everything works great now! We can provide the option to use custom modifiers and it shouldn't be a hassle at all. By the way, I suggest removing the flutter_icons package from the example project because it's a large package and, in my opinion, shouldn't be used in small to medium projects let alone an example + it takes a lot of time for GitHub actions to download and that just wastes minutes...

**Comment by aissat** [comment]
@HatEmU  Thanks for the PR

**Comment by Overman775** [comment]
@HatEmU nice PR 😃 Can you add a description to the Readme file?

File: lib/src/localization.dart [hunk 1]
```diff
@@ -13,6 +13,15 @@ class Localization {
   String path;
   bool useOnlyLangCode;
   final RegExp _replaceArgRegex = RegExp(r'{}');
+  final RegExp _linkKeyMatcher =
+      RegExp(r'(?:@(?:\.[a-z]+)?:(?:[\w\-_|.]+|\([\w\-_|.]+\)))');
+  final RegExp _linkKeyPrefixMatcher = RegExp(r'^@(?:\.([a-z]+))?:');
+  final RegExp _bracketsMatcher = RegExp(r'[()]');
+  final _modifiers = <String, String Function(String)>{
+    'upper': (String val) => val.toUpperCase(),
```
Comment (Overman775) [inline lib/src/localization.dart:21]: Need tests for modifiers

File: lib/src/localization.dart [hunk 2]
```diff
@@ -13,6 +13,15 @@ class Localization {
   String path;
   bool useOnlyLangCode;
   final RegExp _replaceArgRegex = RegExp(r'{}');
+  final RegExp _linkKeyMatcher =
+      RegExp(r'(?:@(?:\.[a-z]+)?:(?:[\w\-_|.]+|\([\w\-_|.]+\)))');
+  final RegExp _linkKeyPrefixMatcher = RegExp(r'^@(?:\.([a-z]+))?:');
+  final RegExp _bracketsMatcher = RegExp(r'[()]');
```
Comment (Overman775) [inline lib/src/localization.dart:19]: brakets work like in vue? `@:(message.foo.bar.baz)`?

Comment (Overman775) [reply lib/src/localization.dart:19]: need test

**Review by Overman775 — CHANGES_REQUESTED** [review]

File: lib/src/localization.dart (hunk 2 above)
Comment (HatEmU) [reply lib/src/localization.dart:19]: Yes, exactly like that. And I've already added tests in my PR.

**Comment by HatEmU** [comment]
> @HatEmU nice PR 😃 Can you add a description to the Readme file?

Alright, I'll get to it right away.

**Review by Overman775 — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -257,6 +257,73 @@ print('day'.plural(21)); // output: 21 день
...
+}
+```
+```dart
+print('dateLogging'.tr(namedArguments: {'currentDate': DateTime.now().toIso8601String()})); //Output: INFO: the date today is 2020-11-27T16:40:42.657.
+```
+
+### Formatting linked translations:
+
+Formatting linked locale messages
+If the language distinguish cases of character, you may need control the case of the linked locale messages. Linked messages can be formatted with modifier @.modifier:key
```
Comment (aissat) [inline README.md:300]:
```suggestion
If the language distinguishes cases of character, you may need to control the case of the linked locale messages. Linked messages can be formatted with modifier `@.modifier:key`
```

File: README.md [hunk 4]
```diff
@@ -257,6 +257,73 @@ print('day'.plural(21)); // output: 21 день
...
+
+The below modifiers are available currently.
+
+- upper: Uppercase all characters in the linked message.
```
Comment (aissat) [inline README.md:304]:
```suggestion
- `upper`: Uppercase all characters in the linked message.
```

File: README.md [hunk 5]
```diff
@@ -257,6 +257,73 @@ print('day'.plural(21)); // output: 21 день
...
+- lower: Lowercase all characters in the linked message.
```
Comment (aissat) [inline README.md:305]:
```suggestion
- `lower`: Lowercase all characters in the linked message.
```

File: README.md [hunk 6]
```diff
@@ -257,6 +257,73 @@ print('day'.plural(21)); // output: 21 день
...
+- capitalize: Capitalize the first character in the linked message.
```
Comment (aissat) [inline README.md:306]:
```suggestion
- `capitalize`: Capitalize the first character in the linked message.
```

**Comment by aissat** [comment]
@HatEmU nice PR 👍  plz check small changes in Readme file

**Comment by HatEmU** [comment]
> @HatEmU nice PR 👍 plz check small changes in Readme file

👍 Thanks for the help!

**Pull request merged**